src/ExtInterface/ReefMod/Domain.jl

```diff
@@ -104,29 +103,24 @@ function load_domain(::Type{ReefModDomain}, fn_path::String, RCP::String)::ReefM
     zones = replace.(zones, "Zone" => "", " " => "")
     site_data[:, :zone_type] .= zones
 
-    timeframe = (2022, 2100)
-
     # This loads cyclone categories, not mortalities, so ignoring for now.
     # cyc_scens = load_cyclones(ReefModDomain, data_files, loc_ids, timeframe)
 
-    # timesteps, location, scenario
-    wave_scens = zeros(length(timeframe[1]:timeframe[2]), nrow(site_data), 1)
-    wave_scens = NamedDimsArray(
-        wave_scens;
-        timesteps=timeframe[1]:timeframe[2],
-        locs=loc_ids,
-        scenarios=[1]
-    )
+    timeframe = (2022, 2100)
+    timeframe_range = timeframe[1]:timeframe[2]
 
-    # timesteps, location, species, scenario
-    cyc_scens = zeros(length(timeframe[1]:timeframe[2]), nrow(site_data), 6, 1)
-    cyc_scens = NamedDimsArray(
-        cyc_scens;
-        timesteps=timeframe[1]:timeframe[2],
-        locs=loc_ids,
-        species=1:6,
-        scenarios=[1]
-    )
+    dim_timesteps = Dim{:timesteps}(timeframe_range)
+    dim_locs = Dim{:locs}(loc_ids)
+    dim_species = Dim{:species}(1:6)
+    dim_scenarios = Dim{:scenarios}([1])
+
+    wave_axlist = (dim_timesteps, dim_locs, dim_scenarios)
+    wave_data = zeros(length(timeframe_range), nrow(site_data), 1)
+    wave_scens::YAXArray{Float64} = YAXArray(wave_axlist, wave_data)
+
+    cyc_axlist = (dim_timesteps, dim_locs, dim_species, dim_scenarios)
+    cyc_data = zeros(length(timeframe_range), nrow(site_data), 6, 1)
+    cyc_scens::YAXArray{Float64} = YAXArray(cyc_axlist, cyc_data)
```

Comment:
Feels like this could be a good case of a generic constructor for all Domain types to use